### DISCUSSION TITLE
Some Ideas for P1P / P1S Integration

### DISCUSSION THREAD

**Discussion by Metallica22** [greghesp/ha-bambulab#237] [Ideas]
Dear Development Team,
I enthusiastically use the existing BambuLab integration in Home Assistant for my P1P and P1S models. 

In my P1P, I have retrofitted the Chamber Fan, while it is built into the P1S by default. 

To further enhance the usage and control of the Chamber Fan and optimize environmental monitoring, I would like to propose the following features:

Chamber Fan Control:
Integrate the ability to control the Chamber Fan for individual and on-demand chamber ventilation.

Automatic Temperature Control:
Implement automatic temperature control based on readings from various temperature sensors (e.g., Xiaomi Temperature Sensor), allowing efficient regulation of the chamber's temperature.

I believe these features would significantly enhance the Home Assistant integration for BambuLab models P1P and P1S, providing users with an even better control experience.
Thank you for your excellent work and considering my suggestions.

Best regards,

Patrick

**Comment by greghesp** [comment]
Hey, 

Not sure I understand the ask here.

Chamber Fan Control already exists, with the ability to turn on/off, or set a speed.

Automatic temperature control is something you would need to build in HA. An integration doesn't provide automations, but rather exposes the information and actions to be able to build an automation in HA

**Comment by AdrianGarside** [comment]
Chamber fan is now present even for P1P which matches how Bambu's apps handle the two printers.
